### PR TITLE
Automatic generation of tests

### PR DESCRIPTION
**Pull request by LucasAlegre**
Solves #36.

### CODE REVIEW

File: tests/test_envs.py [hunk 1]
```diff
@@ -1,142 +1,134 @@
...
+    - observation after first reset are the same
+    - same actions are sampled by the two envs
+    - observations are contained in the observation space
+    - obs, rew, done and info are equals between the two envs
+    """
+    # Don't check rollout equality if it's a nondeterministic environment.
+    if env_spec.nondeterministic is True:
+        return
 
+    env_1 = env_spec.make(disable_env_checker=True)
```
Comment (ffelten) [inline tests/test_envs.py:68]: `mo_gym.make` ?

File: tests/test_envs.py [hunk 2]
```diff
@@ -1,142 +1,134 @@
...
-    env = mo_gym.make("mo-mountaincarcontinuous-v0")
-    env = LinearReward(env)
+all_testing_env_specs = []
+for env_spec in gym.envs.registry.values():
+    if type(env_spec.entry_point) is not str:
+        continue
+    # collect MO Gymnasium envs
+    if env_spec.entry_point.split(".")[0] == "mo_gymnasium":
+        # Ignore highway as they do not deal with the random seed appropriately
+        if not env_spec.id.startswith("mo-highway"):
```
Comment (ffelten) [inline tests/test_envs.py:19]: I think we should still implement tests for these guys (at least for the CI to tell us if they're broken)

**Review by ffelten — APPROVED** [review]
Very cool! Thanks

File: tests/test_envs.py (hunk 2 above)
Comment (LucasAlegre) [reply tests/test_envs.py:19]: I saw on the highway-env repo and they fixed this already, but it is not released on pypi yet. So it should be fixed in their next release

File: tests/test_envs.py (hunk 1 above)
Comment (LucasAlegre) [reply tests/test_envs.py:68]: mo_gym.make simply calls gym.make(disable_env_checker=True), so this is fine.

File: tests/test_envs.py (hunk 2 above)
Comment (LucasAlegre) [reply tests/test_envs.py:19]: I will open an issue for this.

**Pull request merged**